### ISSUE TITLE
Regression in 12.1.0 on scrollable p-table with colspan/rowspan

### ISSUE THREAD

**Issue by marob** [primefaces/primeng#10541]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**

I've based my POC on a PrimeNG showcase:
https://stackblitz.com/edit/primeng-tablecolgroup-demo-qfmbtw?file=src/app/app.component.html

**Current behavior**
When enabling scrollable on a p-table using rowspan or colspan, the UI completely breaks: colspans and rowspans are not taken into account anymore.

This is a regression compared to 12.0.2 that was introduced in 12.1.0-rc.1 (and that is still present in 12.1.0), probably by the commit https://github.com/primefaces/primeng/commit/08b514c71bd1267b3f0cf029b084d0fcc6744fce, that uses flexbox now.

**Expected behavior**
Enabling scrollable shouldn't change how rowspan and colspan are taken into account.

**Minimal reproduction of the problem with instructions**
On https://stackblitz.com/edit/primeng-tablecolgroup-demo-qfmbtw?file=src/app/app.component.html, click the `Enable "scrollable" and break the UI` button to see how the UI breaks (you can try with a small or large viewport).

You can also test by changing the primeng dependency version to 12.0.2 and see that the bug wasn't there (even if there's small UI changes, it's not broken).

**Comment by marob** [comment]
@yigitfindikli Is it possible to reopen this issue? Or should I create a new one?

**Comment by chandan-bcsg** [comment]
> I can currently have a workaround with the following CSS (go back to table instead of flex), but that's not ideal:
> 
> ```css
> .p-datatable-grouped-header thead {
> 	display: table-header-group !important;
> }
> .p-datatable-grouped-header tbody {
> 	display: table-row-group !important;
> }
> .p-datatable-grouped-header tbody tr {
> 	display: table-row !important;
> }
> .p-datatable-grouped-header tbody tr td {
> 	display: table-cell !important;
> }
> ```

It fixes the Row span issue but adds an unwanted horizontal scrollbar .

**Comment by marob** [comment]
This is (finally) fixed in 14.2.1
